File: frontend/src/Meta.js
import Config from './Config';

class Meta {
  constructor(meta) {
    this.alias = meta.alias;
    this.name = meta.name;
    this.hash = meta.hash;
    this.slug = meta.slug;
    this.type = meta.type;
    this.path = meta.path;
    this.size = meta.size;
    this.date = meta.date;
    this.bytesReceived = meta.bytesReceived;
  }

  obj = () => {
    return {
      alias: this.alias,
      name: this.name,
      hash: this.hash,
      slug: this.slug,
      type: this.type,
      path: this.path,
      size: this.size,
      date: this.date,
      bytesReceived: this.bytesReceived
    }
  }
}

export default Meta;

File: backend/internal/domain/file/meta.go
package file

import (
	"time"

	"github.com/zqz/web/backend/internal/models"
)

type File struct {
	models.File

	BytesReceived int
	Thumbnail     string `json:"thumbnail,omitempty"`
}

type Meta struct {
	ID            int       `json:"-"`
	Alias         string    `json:"alias,omitempty"`
	Name          string    `json:"name"`
	Hash          string    `json:"hash"`
	Slug          string    `json:"slug"`
	ContentType   string    `json:"type"`
	Path          string    `json:"path,omitempty"`
	Size          int       `json:"size"`
	Date          time.Time `json:"date"`
	BytesReceived int       `json:"bytes_received,omitempty"`
	Private       bool      `json:"private"`
	Comment       string    `json:"comment"`
	UserID        int       `json:"-"`

	Thumbnail string `json:"thumbnail,omitempty"`
}

func (m File) Finished() bool {
	return m.Size == m.BytesReceived
}
